modules/__tests__/Record-test.js

```diff
@@ -118,6 +118,46 @@ describe('Record', () => {
     expect(entity.equals(equal)).toBe(true);
   });
 
+  class Embedded extends Record {
+    name: string = 'name';
+    age: number = 1;
+    entity: Entity = new Entity();
+    date: Date = new Date();
+    obj: Object = {foo: 'bar'};
+  }
+  
+  it('should be serializable with embedded dataclass', () => {
+    let embedded = new Embedded({
+      date: new Date('1996-12-17T03:24:00')
+    });
+    let raw = {
+      name: "name",
+      age: 1,
+      entity: {
+        someString: "default string",
+        someNum: 0.134,
+        someBool: true,
+        someNullable: null
+      },
+      date: "1996-12-16T19:24:00.000Z",
+      obj: {
+        foo: "bar"
+      }
+    };
+    expect(JSON.stringify(embedded)).toBe(JSON.stringify(raw));
+  })
+
+  // FIXME: equals cannot compare dataclass with value other than primitive value
```

Comment:
Thanks for putting this test here. I have an idea for fixing it, can be done after merging the PR.